_posts/2018-11-15-Rust-survey-2018.md

```diff
@@ -0,0 +1,168 @@
+
+![How long did it take to be productive](/images/2018-11-RustSurvey/5-How_long_did_it_take_to_be_productive.png)
+
+Over 40% of Rust users felt productive in Rust in less than a month of use, and over 70% felt productive in their first year. Unfortunately, there is a noticeable struggle among users, as over 22% do not yet feel productive.
+
+![How long have you been unproductive](/images/2018-11-RustSurvey/5a-How_long_have_you_been_unproductive.png)
+
+Looking closer at these users who feel unproductive in Rust, only about 25% are in their first month of use. The challenge here is to find ways to help bridge users to productivity so they don't get stuck.
+
+## **How much do you use Rust?**
+
+![Size of summed Rust projects](/images/2018-11-RustSurvey/6-Size_of_summed_Rust_projects.png)
+
+Rust projects are continuing to trend to larger sizes, with larger overall investments. Medium to large investments in Rust (those totally over 10k and 100k lines of code respectively) have grown from 8.9% in 2016, to 16% in 2017, to **23%** this year. 
+
+![How often do you use Rust](/images/2018-11-RustSurvey/7-How_often_use_Rust.png)
+
+We’ve also seen a growth in Rust regular usage. Up from 17.5% last year, Rust daily usage is now nearly a quarter of users. In total, Rust weekly total usage has risen from 60.8% to 66.4%.
+
+## **Rust expertise**
+
+![How would you rate your Rust expertise](/images/2018-11-RustSurvey/8-How_you_rate_your_Rust_expertise.png)
+
+Rather than being a simple curve, Rust expertise has two peaks: one around a "3", and another at "7", showing that users tend to see themselves as just above beginner or experienced without necessarily being expert.
+
+![How difficult are Rust concepts](/images/2018-11-RustSurvey/9-How_difficult_are_Rust_concepts.png)
+
+Rust users generally felt that Enums and Cargo were largely easy concepts; followed by Iterators, Modules, and Traits. More challenging concepts of Trait Bounds and Unsafe came next. Lastly, the most challenging concepts were Macros, Ownership & Borrowing, and Lifetimes. These challenges match closely to feedback we’ve heard in years past and continue to be a focus of continued productivity improvements like NLL and the continued macro system improvements. 
+
+![What programming languages are you familiar with](/images/2018-11-RustSurvey/22-Programming_language_familiarity.png)
+
+Humorously, we see that Rust isn't actually the top programming language that users were familiar with. Instead, it pulled in a 2nd place behind Python.
+
+## **Rust toolchain**
+
+![Which Rust version do you use](/images/2018-11-RustSurvey/10-Which_Rust_version.png)
+
+We’re seeing similar numbers in users of the current stable release since last year. Perhaps surprisingly, we’re continuing to see a rise in the number of users who use the Nightly compiler in their workflow. For the second year in a row, Nightly usage has continued to rise, and is now over 56% (up from 51.6% of last year).
```

Comment:
Did we happen to ask why folks use nightly?

Reply:
We did! I put some examples in the latest draft.